Sources/Jaeger/Sampling/ProbabilisticSampler.swift

```diff
@@ -0,0 +1,32 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Jaeger Client Swift open source project
+//
+// Copyright (c) 2020 Moritz Lang and the Jaeger Client Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import W3CTraceContext
+
+public struct ProbabilisticSampler: Sampler {
+    private let probability: Double
+    private let lowUpperBound: UInt64
+
+    public init(probability: Double = 0.001) {
+        precondition(probability > 0 && probability < 1, "The sampling rate must be greater than 0.0 and less than 1.0")
```

Comment:
```suggestion
        precondition(probability > 0 && probability <= 1, "The sampling rate must be greater than 0.0 and less than 1.0")
```

Not sure there's a need to forbid the 0.0 and 1.0 here; sure constant strategy is better but this is still correct?

Or it this directly because of how other impls do it?

Reply:
I copied this bit from the Java client: https://github.com/jaegertracing/jaeger-client-java/blob/dfb9d8b9dc849f2e6f880b44b26fb77dfc894a02/jaeger-core/src/main/java/io/jaegertracing/internal/samplers/ProbabilisticSampler.java#L37

I think it's fair to fail here because it will be reflected in the documentation. I think the exception message could be improved though:

```suggestion
        precondition(probability > 0 && probability < 1, "The probability must be greater than 0.0 and less than 1.0. Please use ConstantSampler instead.")
```

Reply:
sampling rate sounds good to me tbh

Okey let's go with > 0 and  < 1, thx for linking!

Reply:
Okay, I like sampling rate as well. Let's use it instead of `probability` in the sampler as well 👍